### DISCUSSION TITLE
responsive tabs and maximum number of tabs

### DISCUSSION THREAD

**Discussion by charleneyjh1** [twilio-labs/paste#2223] [Q&A] [answered]
Hi team, 

I can across a use case where I might need to use 8 tabs to show that Twilio supports multiple coding languages. I saw that the Figma component only have 7 tabs maximun, so I'm wondering if that's an API restriction?
<img width="827" alt="Screen Shot 2022-02-17 at 10 16 23 AM" src="https://user-images.githubusercontent.com/56697620/154535619-2395a1eb-7f42-410e-be22-2b2339306112.png">

Another issue that I came across is the spacing/responsiveness of the tabs. The current layout (1200px wide screen) can only fit 6 tabs. Is there anything we can do to try to fit more tabs in it? For example, using a dropdown to show more tabs are available when screen sizes are smaller. Is this something we can do? Any guidance would help, thanks!

<img width="619" alt="Screen Shot 2022-02-17 at 10 16 16 AM" src="https://user-images.githubusercontent.com/56697620/154535696-104e546a-5003-48e3-a0d4-96e5bc3a885c.png">

**Comment by TheSisb** [comment, marked as answer]
Hi @charleneyjh1,

There's no API restriction on the number of tabs in code. I believe the Figma restriction was to keep designers mindful of different browser resolutions, as having too many horizontal tabs can cause a bad experience on smaller devices.

Some suggestions would be to consider using vertical tabs in your design or a [Select component](https://paste.twilio.design/components/select) to make the selection more responsive friendly.

**Reply by charleneyjh1** [reply]
Thank you Shabi, I'll cut down the number of tabs.

**Reply by stern-shawn** [reply]
Hi @TheSisb ! Chiming in here since we're running into a similar issue on the new Docs platform; let me know if it'd be more reasonable to start a new discussion 🙏 

We have some... not great behavior with tabs on mobile views, even with say 5 tabs and relatively short titles:

![Screenshot_20230901-090207](https://github.com/twilio-labs/paste/assets/10067791/9426d0e5-1d6f-4f4d-9efc-36a27eee20ca)
([Doc itself](https://www.twilio.com/docs/twilio-cli/quickstart), these are live now thanks to Paste 🙌 )

A couple ideas we have are to make the tabs bar scrollable, similar to a [pattern I found in the MUI library](https://mui.com/material-ui/react-tabs/#scrollable-tabs). I have a rudimentary version working (sans auto-scrolling as the user navigates with left/right arrow keys, etc, but the scrolling potential is there) as seen here:

https://github.com/twilio-labs/paste/assets/10067791/af4ccc74-b0eb-44c6-990e-3a126c92f7d1

We've also tossed around the notion of rendering a select/combobox component at mobile breakpoints (similar to how we're rendering a menu as opposed to the side-by-side table of contents on tablet/mobile), which I see you suggested! My question there is, how would we approach implementing that while also keeping accessibility in mind? Would it be accessible to have a structure like this, and leverage the [Tabs state hook](https://paste.twilio.design/components/tabs#using-state-hooks) + Select events to manage which is selected?

```jsx
<Tabs>
  <Select /> // Might be outside of the Tabs, ignore the bad pseudocode
  <TabPanels>
    ...
  </TabPanels>
</Tabs>
```

I think I understand how to do this and make it _work_, but I'm concerned that the pattern may not make sense to a reader, especially one using assistive technology. Any help would be much appreciated 😄

**Reply by TheSisb** [reply]
Hi @stern-shawn,

Really comprehensive question and though process. Can you make a new thread please? I don't want it to get lost in the sub-comment section of this one if possible.

**Reply by stern-shawn** [reply]
Of course, much appreciated. I've created a new discussion at #3508!
